src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubListener.cs

```diff
@@ -0,0 +1,122 @@
+using Microsoft.Azure.WebJobs.Host.Triggers;
+using Microsoft.Azure.WebJobs.Host.Listeners;
+using Microsoft.Azure.WebJobs.Host.Executors;
+
+namespace Microsoft.Azure.WebJobs.ServiceBus
+{
+    // Created from the EventHubTrigger attribute to listen on the EventHub. 
+    internal sealed class EventHubListener : IListener, IEventProcessor, IEventProcessorFactory
+    {
+        private ITriggeredFunctionExecutor _executor;
+        private readonly EventProcessorHost _eventListener;
+        private readonly bool _singleDispatch;
+        private readonly EventProcessorOptions _options;
+
+        public EventHubListener(ITriggeredFunctionExecutor executor, EventProcessorHost eventListener, EventProcessorOptions options, bool single)
+        {
+            this._executor = executor;
+            this._eventListener = eventListener;
+            this._singleDispatch = single;
+            this._options = options;
+        }
+
+        public void Cancel()
+        {
+        }
+
+        public void Dispose()
+        {
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {            
+            return _eventListener.RegisterEventProcessorFactoryAsync(this, _options);
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            return _eventListener.UnregisterEventProcessorAsync();
+        }
+
+
+        async Task IEventProcessor.OpenAsync(PartitionContext context)
+        {
+            // Begin listener 
+        }
+
+        async Task IEventProcessor.ProcessEventsAsync(PartitionContext context, IEnumerable<EventData> messages)
+        {
+            EventHubTriggerInput value = new EventHubTriggerInput
+            {
+                _events = messages.ToArray(),
+                _context = context
+            };
+
+            // Single dispatch 
+            if (_singleDispatch)
+            {
+                int len = value._events.Length;
+
+                Task[] dispatches = new Task[len];
+                for (int i = 0; i < len; i++)
+                {
+                    TriggeredFunctionData input = new TriggeredFunctionData
+                    {
+                        ParentId = null,
+                        TriggerValue = value.GetSingleEvent(i)
+                    };
+                    dispatches[i] = _executor.TryExecuteAsync(input, CancellationToken.None);
+                }
+
+                // Drain the whole batch before taking more work
+                await Task.WhenAll(dispatches);
+            }
+            else
+            {
+                // Batch dispatch
+
+                TriggeredFunctionData input = new TriggeredFunctionData
+                {
+                    ParentId = null,
+                    TriggerValue = value
+                };
+
+                FunctionResult result = await _executor.TryExecuteAsync(input, CancellationToken.None);
+            }
+
+            await context.CheckpointAsync();
```

Comment:
Probably should only checkpoint if the function succeeded? TryExecuteAsync won't throw if the function fails, it just returns a failed FunctionResult